src/components/Dashboard/TodoList.vue

```diff
@@ -796,6 +833,70 @@ export default class TodoList extends Mixins(AllowableActionsMixin, DateMixin, E
     }
   }
 
+  /** check if the item is actually affiliation invite todo; (not a regular filing item). */
+  isAffiliationInvitation (item): boolean {
+    return item.status === FilingStatus.AFFILIATION_INVITATION_PENDING
+  }
+
+  authorizeAffiliationInvitation (isAuthorized, affiliationInvitationTodo): void {
+    AuthServices.authorizeAffiliationInvitation(
+      this.getAuthApiUrl,
+      this.getIdentifier,
+      affiliationInvitationTodo.affiliationId,
+      isAuthorized
+    )
+      .then()
+      .catch(err => {
+        // eslint-disable-line no-console
+        console.log('failed the call for authorization of affiliation invitation', err)
+      })
+  }
+
+  _buildTodoItemIfFromAffiliationInvitation (affiliationInvitation) {
+    const newTodo: TodoItemIF = {
+      draftTitle: undefined,
+      enabled: true,
+      filingId: -1, // not a filing
+      name: undefined,
+      order: 0,
+      subtitle: `From: ${affiliationInvitation.fromOrg.name}`,
+      status: FilingStatus.AFFILIATION_INVITATION_PENDING,
+      title: 'Request for authorization to mange this business',
+      affiliationInvitationDetails: {
+        id: affiliationInvitation.id,
+        fromOrgName: affiliationInvitation.fromOrg.name,
+        additionalMessage: affiliationInvitation.additionalMessage
+      }
+    }
+
+    return newTodo
+  }
+
+  /** Loads Org Affiliations invitation todo **/
+  async loadAffiliationInvitationsTodo () {
```

Comment:
Can this follow the existing design of the Legal API creating any todo tasks?

Reply:
For sure, but can you give me a bit more details what should be changed and/or refer me to the specific parts of the code you want my code to be similar to?

Are you possibly referring to the loadTodoItem method and adding "Affiliation Invite" as a task first ( and to create a new FilingType) or to something else ?

Thanks!

Reply:
I believe all todo items come from the Legal API. The Legal API dynamically creates todo tasks as needed (eg, an Annual Report is due).

Here, the Legal API would add a todo task if an invitation needs to be approved. The TodoList would simply display it (may need small TodoList updates to handle display).

@argush3 @vysakh-menon-aot Could one of you link to the code in Legal API where todo tasks are created?

Reply:
I am not fully familiar with all the APIs that exist as part of the system. I did work on Auth API, and currently all the data in the backend will be stored in the Auth API. 
I am assuming these two are not the same backend projects. And if so, I would prefer to avoid adding proxy backend calls.

I believe that avoiding proxy call would reduce complexity, improve maintenance and simplicity of understanding where data is coming from.

Also, I tried to put the call to load affiliation invitations at the place where data is getting loaded into the control, as my understanding is that they are not to be part of any filings or anything else other then todo reminder on that specific page.

That being said, if it is mandatory to go through Legal API to provide todo data I am not against it, just think that approach to getting data from the source API might be more straightforward.

Thank you!

Reply:
Maybe it would make sense to discuss this offline and see what is the desired approach for this specific case. 

I could get better understanding of the API relations to this frontend and their role and I can share why I chose this approach. 

I am open to advice and can update my approach/code to fit the current repo both in style and data flow.

Thanks!

Reply:
I'm hoping Argus or Vysakh can provide a link for you to look at. We can discuss next week if needed.

The basic principle is that the UI should be as dumb as possible.

Reply:
hey Sev thats a good point, but I think from a backend perspective it really doesn't make sense to combine auth/lear data in the legal-api response. In general I think the apis should be as decoupled as possible and the UI can decide what it needs from each one.  I think that makes it simpler when designs change or new use cases come up and makes a lot more sense from a 3rd party api perspective too

Reply:
OK, let's try to have the Todo List fetch data from both APIs...

1. do affiliation invitations have a status (eg, new)?
2. do affiliation invitations block other filings, ie, only allow 1 active filing active at a time?
3. do any other scenarios block affiliation invitations,ie, do we need `if (isAllowed(affilliation_invitation))`?

Reply:
OK, thanks. Resolving. (I have made other comments to make this fit more cleanly with the existing code.)

Reply:
Agree with @kialj876 
1) not that it would matter (only pending status, for specific type of the affiliation invitation is displayed)
2) does not block any filling, it is just something owner has as a todo item while invite is 'pending'
3) no scenarios that would block affiliation invitations (affiliation invitation is basically a request from someone to manage the business; access to the managing org)